pkg/server/server.go

```diff
@@ -2055,6 +2029,35 @@ func (s *topLevelServer) PreStart(ctx context.Context) error {
 		}
 	}
 
+	// Connect the HTTP endpoints. This also wraps the privileged HTTP
+	// endpoints served by gwMux by the HTTP cookie authentication
+	// check.
+	// NB: This must occur after sqlServer.preStart() which initializes
+	// the cluster version from storage as the http auth server relies on
+	// the cluster version being initialized.
```

Comment:
Is it worth noting that in `PreStart` for the top level server it seems like we actually do init the cluster version prior to sql server and http routes setup, (which is why the system tenant / non multitenant servers weren't encountering the errors seen before the ordering fix): https://github.com/cockroachdb/cockroach/blob/cc8f6678e08b217c5926b6e8e82da774168c1739/pkg/server/server.go#L1513-L1515.

I don't really have context on what that initial cluster version load is doing and how it differs from the one in the sql server though, so perhaps this comment is still valid. If so ignore this comment 👍

Reply:
I left the comment as-is for now until I can untangle the full situation.